content/blog/2019/10/2019-10-03-jcasc-phase1-VSCode-Case-Study.adoc

```diff
@@ -0,0 +1,92 @@
+---
+layout: post
+title: "JCasC Community Bridge Dev Tools - Phase 1"
+opengraph:
+  image: /images/logos/JCasC/JCasC.png
+tags:
+- community-bridge
+- JCasC
+- VSCode
+- community
+author: sladyn98
+---
+
+=== Community Bridge Intoruction
+link:https://communitybridge.org/[Community Bridge] is an initiative by the Linux Foundation to accelerate the adoption , innovation and sustainability of open source projects .I came across this initiative in a blog post. I had been contributing to Jenkins at the time and decided to have a chat with Oleg-Nenashev and Tracy Miranda regarding the possibility of a project under the Community Bridge initiative.Fortunately for me JCasC ( Jenkins Configuration as Code ) had the mentors as well as the project idea in place to start a project.After a few regular meetings we ironed out the details of the programme and on August 7th I began with my journey!
+
+=== JCasC Developer Tools — JSON Schema
+
+JSON files when submitted to a server undergo a validation to determine whether the values and the format are correct and that they conform to a well defined schema, this schema is known a JSON Schema.A YAML file can also be validated using a JSON Schema.
+The main premise of JCasC is to load YAML files written by developers into the Jenkins instance.An example of a jenkins YAML file is:
+
+[source, yaml]
+---
+Jenkins:
+ systemMessage: “Hello World”
+ numExecutors:2
+ ---
+
+The above YAML configuration will configure Jenkins to display a message Hello world with the number of executors set to two.In order to validate the schema we have a schema. This schema is written using jelly files (Executable XML files) and currently it is not a valid schema.The first phase of the project is based around rewriting the schema generation in java and developing a better test framework for it, because currently the schema is not testable.
```

Comment:
```suggestion
The above YAML configuration will configure Jenkins to display a message `Hello world` with the number of executors set to two. In order to validate the YAML we have a schema. This schema is written using jelly files (Executable XML files) and currently it is not a valid schema. The first phase of the project is based around rewriting the schema generation to java and developing a better test framework for it, because currently the schema is not testable.
```